frontend/src/pages/[user]/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+import { SWRConfig } from 'swr'
+import useSWRInfinite, { unstable_serialize } from 'swr/infinite'
+
+import Page from '@components/Page'
+import CardsGrid, {
+  getKey,
+  gridFetcher,
+  Project,
+  useUpdateBeforeReachingLimit,
+} from '@components/ProjectCardGrid'
+import { userExists } from '@utils/giteaApi'
+import { GetServerSideProps } from 'next'
+
+export const getServerSideProps: GetServerSideProps = async ({ params }) => {
+  const username = params.user as string
+  const exists = await userExists(username)
+  if (!exists) {
+    return {
+      notFound: true,
+    }
+  }
+
+  const q = {
+    query: '*',
+    filter: `ownerName = ${username}`,
+  }
+
+  const hits = await gridFetcher(q)
+
+  return {
+    props: {
+      swrFallback: {
+        [unstable_serialize(getKey(q))]: hits,
+      },
+      username,
+    },
+  }
+}
+
+interface UserPageProps {
+  username: string
+  swrFallback: Record<string, Project[]>
+}
+
+const UserPage = ({ swrFallback, username }: UserPageProps) => {
+  return (
+    <SWRConfig value={{ fallback: swrFallback }}>
+      <User username={username} />
+    </SWRConfig>
+  )
+}
+
+const User = ({ username }: { username: string }) => {
```

Comment:
Please separate out the props as a type, even if it's just one.  Maybe there's an eslint rule for this?